db/mapper.go

```diff
@@ -41,21 +43,45 @@ func (um utxoMapper) GetUTXO(ctx sdk.Context, position types.Position) types.UTX
 	return utxo
 }
 
+// Returns all the UTXOs owned by an address.
+// Returns empty slice if no UTXO exists for the address.
+func (um utxoMapper) GetAllUTXOsForAddress(ctx sdk.Context, addr common.Address) []types.UTXO {
+	store := ctx.KVStore(um.contextKey)
+	iterator := sdk.KVStorePrefixIterator(store, addr.Bytes())
+	utxos := make([]types.UTXO, 0)
+
+	for ; iterator.Valid(); iterator.Next() {
+		utxo := um.decodeUTXO(iterator.Value())
+		utxos = append(utxos, utxo)
+	}
+	iterator.Close()
+
+	return utxos
+}
+
 // Adds the UTXO to the mapper
 func (um utxoMapper) AddUTXO(ctx sdk.Context, utxo types.UTXO) {
 	position := utxo.GetPosition()
-	pos := um.encodePosition(position)
-
+	address := utxo.GetAddress()
 	store := ctx.KVStore(um.contextKey)
+
+	key := um.getKeyFromAddressAndPosition(address, position)
 	bz := um.encodeUTXO(utxo)
-	store.Set(pos, bz)
+	store.Set(key, bz)
 }
 
-// Deletes UTXO corresponding to the position from mapping
-func (um utxoMapper) DeleteUTXO(ctx sdk.Context, position types.Position) {
+// Deletes UTXO corresponding to address + position from mapping
+func (um utxoMapper) DeleteUTXO(ctx sdk.Context, addr common.Address, position types.Position) {
 	store := ctx.KVStore(um.contextKey)
+	key := um.getKeyFromAddressAndPosition(addr, position)
+	store.Delete(key)
+}
+
+// (<address> + <encoded position>) forms the unique key that maps to an UTXO.
+func (um utxoMapper) getKeyFromAddressAndPosition(address common.Address, position types.Position) []byte {
```

Comment:
I think this can be renamed to getKey. Looking at docstring above and required params will make it clear we construct from address and position

Reply:
how about `constructKey` since this is a helper function? The "get" verb can be misleading since there is also a "GetUTXO" method.

Reply:
sure